postgres/datadog_checks/postgres/connections.py

```diff
@@ -218,16 +205,11 @@ def _terminate_connection_unsafe(self, dbname: str) -> bool:
 
         return True
 
-    def get_main_db_pool(self, max_pool_conn_size: int = 3):
+    @contextlib.contextmanager
+    def get_main_db(self):
         """
-        Returns a memoized, persistent psycopg connection pool to `self.dbname`.
-        Is meant to be shared across multiple threads, and opens a preconfigured max number of connections.
+        Returns a memoized, persistent psycopg connection to `self.dbname`.
         :return: a psycopg connection
         """
-        conn = self._get_connection_pool(
-            dbname=self._config.dbname,
-            ttl_ms=self._config.idle_connection_timeout,
-            max_pool_size=max_pool_conn_size,
-            persistent=True,
-        )
-        return conn
+        with self.get_connection(self._config.dbname, self._config.idle_connection_timeout, persistent=True) as db:
+            yield db
```

Comment:
noob question, but `get_connection` yields a db object wrapped with a context manager, so can't we just return it here? why does this method also need to be a `contextmanager`?

Reply:
yeah that actually works. let me update